draft-ietf-tls-wkech.xml

```diff
@@ -345,120 +342,147 @@ Client <-->|                    | |                    | facing
-                only octets from the ABNF set ALPHA / DIGIT / "-" / "_".
-            </t>
-            <t>
-                alias: The value MUST be a DNS name
-        that could be used as the TargetName of an HTTPS resource record.
-        This indicates that the backend
-        is hosted on the same endpoints as this target, and is equivalent to
-        an HTTPS AliasMode record.
-        The ZF might implement this directive
-        by publishing an AliasMode record, publishing a CNAME record, copying
-        HTTPS records from the target zone, or fetching
-        https://cfs.example.com/.well-known/origin-svcb" (if it exists).
-        In this case, the regeninterval indicates that backend does
-        not plan to change the content at the URL for at least that number of
-        seconds.
-                If an alias entry is present then any ech and port entries
-                (if also present) MUST be ignored.
-        </t>
-            <t>
-                params: A JSON Dictionary representing the SVCB SvcParams.
-                Each key in the dictionary is a string containing a registered
-                SvcParamKey name (e.g. "ipv6hint") or a SvcParamKey in generic form (e.g.,
-                "key65528").
-                <list>
-                    <t>
-                        For single-valued SvcParams (e.g., "ech"), the value is a JSON String.
-                        A JSON String is a sequence of Unicode codepoints, while a
-                        SvcParam's presentation value is a sequence of octets, so each
-                        value octet is stored as a single Unicode codepoint <xref
-                        target="ISOMORPHIC-DECODE"/>.  In almost all cases, this is
-                        equivalent to the ordinary string representation of the presentation
-                        value.
-                    </t>
-                    <t>
-                        For list-valued SvcParams (e.g., "alpn"), the value is a JSON Array
-                        of Strings.  Each String represents an octet sequence, as in the
-                        single-value case.
-                    </t>
-                </list>
-            </t>
-        </list>
-        An empty endpoint object corresponds to an HTTPS record with inferred
-        SvcPriority, TargetName=".", and no ECH support.  An empty record of
-        this kind can be useful as a simple way to make use of the HTTPS RR
+	with an "endpoints" key that contains an array of objects.
+        All other keys MUST be ignored.
+        The length of the list implies restrictions and meaning to the content
+        of the entries:
+    <list style="symbols">
+      <t>
+        An empty endpoints array corresponds to an HTTPS record with inferred
+        SvcPriority, the TargetName equal to ".", and no ECH support.  This can
+        can be useful as a simple way to make use of the HTTPS RR
```

Comment:
I don't think this is what we want.  An empty array should mean "delete all HTTPS records".  An array containing an empty object should mean this.

Reply:
Makes sense; added a new commit to do this.